lib/db/DB.ts

```diff
@@ -40,14 +40,16 @@ class DB {
    */
   public init = async (network = XuNetwork.SimNet, initDb = false): Promise<void> => {
     this.models = await this.loadModels();
-    let newDb = !this.storage;
-    if (this.storage) {
+    let newDb = true;
+    if (this.storage && this.storage !== ':memory:') {
```

Comment:
Since we're adding new functionality here - what do you think about extracting it from `init` to something like `shouldCreateNewDB` and unit testing it?

Reply:
That's fine with me, I'll do that.

Reply:
Done and I actually refactored it a bit more (we don't always need to make this check for one thing, only when `initDb` is true). It's in a separate commit so you can easily see the changes.

Reply:
Perfect, looks much better now.